### PR TITLE
Add sceKernelWaitSignal and sceKernelSendSignal

### PR DESCRIPTION
**Pull request by sunho**
While I was emulating libult in Vita3k, I found these functions are used internally to implement many sync data structures.

### CODE REVIEW

**Comment by sunho** [comment]
Not fully reversed, but I think it's better than nothing...

File: include/psp2/kernel/threadmgr.h [hunk 1]
```diff
@@ -1135,6 +1135,34 @@ int sceKernelDeleteLwCond(SceKernelLwCondWork *pWork);
 int sceKernelSignalLwCond(SceKernelLwCondWork *pWork);
 int sceKernelWaitLwCond(SceKernelLwCondWork *pWork,  unsigned int *pTimeout);
 
+typedef struct SceKernelWaitSignalResult {
```
Comment (Rinnegatamante) [inline include/psp2/kernel/threadmgr.h:1138]: Where is this struct used?

Comment (sunho) [reply include/psp2/kernel/threadmgr.h:1138]: Right below in SceKernelWaitSignalOptParam.
It was used like below in sceult.
```c
SceKernelWaitSignalResult res;
SceKernelWaitSignalOptParam param;
param.result = &res;
sceKernelWaitSignal(0, 0, 0, &param); 
if (res.dret != 0) {
   // fail
}
```

Comment (d3m3vilurr) [reply include/psp2/kernel/threadmgr.h:1138]: an another question, why return field has `d`?

Comment (sunho) [reply include/psp2/kernel/threadmgr.h:1138]: I don't know. It's from sceult assert message. (The message was "dret == 0" iirc) Maybe it's unix convention. I'm seeing some unix people naming error variable as dret.

Comment (d3m3vilurr) [reply include/psp2/kernel/threadmgr.h:1138]: imo, it wouldn't be unix convention. just meaning of the decimal or something.
but i see that.

File: include/psp2/kernel/threadmgr.h [hunk 2]
```diff
@@ -1135,6 +1135,34 @@ int sceKernelDeleteLwCond(SceKernelLwCondWork *pWork);
...
+    SceUID thid;
+    SceUInt32 dret;
+}SceKernelWaitSignalResult;
```
Comment (d3m3vilurr) [inline include/psp2/kernel/threadmgr.h:1141]: could you add space after `}`?

File: include/psp2/kernel/threadmgr.h (hunk 1 above)
Comment (sunho) [reply include/psp2/kernel/threadmgr.h]: I looked into this again. It turned out the fourth argument doesn't exist. It was just a random coincidence that an important struct used by libult was inside fourth register and not cleared.

**Comment by sunho** [comment]
I updated header to be more correct.

The below sample program demonstrates the behavior of signal api.

```c
#include <psp2/kernel/threadmgr.h>
#include <psp2/kernel/processmgr.h>
#include <stdio.h>


int threadFunc(SceSize args, void *argp) {
    printf("Sleep waiter thread\n");
    sceKernelDelayThread(1*1000000); 
    printf("First wait start\n");
    sceKernelWaitSignal(0,0,0);
    printf("First wait success\n");
    printf("Second wait start\n");
    sceKernelWaitSignal(0,0,0);
    printf("Second wait success\n");
}

int main(int argc, char *argv[]) {
    int thid = sceKernelCreateThread("my_thread", threadFunc, 0x10000100, 0x10000, 0, 0, NULL);
    if (thid >= 0)
        sceKernelStartThread(thid, 0, 0);
    printf("First send\n");
    sceKernelSendSignal(thid);
    printf("Sleep sender thread\n");
    sceKernelDelayThread(3*1000000);
    printf("Second send\n");
    sceKernelSendSignal(thid);
    printf("Third send\n");
    int ret = sceKernelSendSignal(thid);
    printf("Third send error %d \n", ret);
    printf("Fourth send\n");
    ret = sceKernelSendSignal(thid);
    printf("Fourth send error %d \n", ret);
    sceKernelExitProcess(0);
    return 0;
}
```

Here's the output.

```
First send
Sleep waiter thread
Sleep sender thread
First wait start
First wait success
Second wait start
Second send
Third send
Second wait success
Third send error 0
Fourth send
Fourth send error -2147319519
```

You can see that signal is behaving like a semaphore with limit 1.

**Pull request merged**